### PR TITLE
Add C style comment check

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
Discussion on Slack indicated that people were very happy to enforce C++ style single-line commenting only (`//`), and disallow C style commenting, or multiline commenting (`/*   */`).  I was happy to see that response.  I fixed the dozen or so places where C style comments had trickled in and added a custom_check to enforce that we don't get any new multiline comments.  This should be able to drop right in assuming it is clean from GHA.

Slack conversation highlights from @mbadams5:
> I much prefer C++-style comments since you can see the impact of commenting out multiple lines of code in Git. With C-style comments, you only see the beginning and ending line changes.
> I will also add that i’ve had it cause really subtle merge issues before because of that exact issue. The inner lines of the C-style comments were not touched and thus could be merged with new code without conflicts. With C++-style comments, you should always get a merge conflict.

### CODE REVIEW

**Comment by Myoldmopar** [comment]
Build warning on Mac is from the Unitary merge and is fixed in another branch.  Otherwise CI is **super** clean.

File: scripts/dev/check_for_c_style_comments.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from json import dumps
+from pathlib import Path
+from sys import exit
+
+dirs = [
+    Path(__file__).resolve().parent.parent.parent / 'src' / 'EnergyPlus',
+    Path(__file__).resolve().parent.parent.parent / 'tst' / 'EnergyPlus' / 'unit'
+]
+total_c_comments = 0
+for d in dirs:
```
Comment (Myoldmopar) [inline scripts/dev/check_for_c_style_comments.py:65]: New script searches in the source and test directories looking semi-intelligently for C style comments.

File: scripts/dev/custom_check.sh [hunk 2]
```diff
@@ -16,5 +16,6 @@ python3 "$REPO_ROOT"/scripts/dev/check_for_bom_in_idfs.py || EXIT_STATUS=$?
 python3 "$REPO_ROOT"/scripts/dev/verify_cmake_dirs.py || EXIT_STATUS=$?
 python3 "$REPO_ROOT"/scripts/dev/find_included_cc_files.py || EXIT_STATUS=$?
 python3 "$REPO_ROOT"/scripts/dev/analyze_state.py || EXIT_STATUS=$?
+python3 "$REPO_ROOT"/scripts/dev/check_for_c_style_comments.py || EXIT_STATUS=$?
```
Comment (Myoldmopar) [inline scripts/dev/custom_check.sh:19]: Added the new script to the custom_check so it is executed by GHA on every commit.

File: src/EnergyPlus/AirflowNetworkBalanceManager.cc [hunk 3]
```diff
@@ -5153,65 +5153,64 @@ namespace AirflowNetworkBalanceManager {
...
-        "AirflowNetwork:Distribution:Component:TerminalUnit"; if (state.dataAirflowNetwork->AirflowNetworkCompData(j).CompTypeNum ==
-        iComponentTypeNum::HEX) CompName(2) = "AirflowNetwork:Distribution:Component:HeatExchanger"; ShowSevereError(state, std::string{RoutineName} +
-        "Duplicated component names are found = " + state.dataAirflowNetwork->AirflowNetworkCompData(i).Name); ShowContinueError(state, "A
-        unique component name is required in both objects " + CompName(1) + " and " + CompName(2)); ErrorsFound = true;
-                    }
-                }
-            }
-        }
-        */
+        //        for (i = 1; i <= state.dataAirflowNetwork->AirflowNetworkNumOfComps; ++i) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetworkBalanceManager.cc:5156]: These are the bulkiest changes, there were some large chunks of code commented out with C style comments, and they are now single line comments.  No functional change, and this is the way any modern IDE would comment a block of code.  Also, if you ended up uncommenting a block of code that was commented in this way, and made a commit, the lines changed would not include these lines, which could make tracking down regressions more difficult. Thanks for that heads up @mbadams5, I hadn't thought about that key aspect.

File: scripts/dev/license-check.py [hunk 4]
```diff
@@ -101,7 +101,8 @@
 # Check files
 python_file_license_success = True
 patterns = [r'.*third_party.*', r'^\.(\\|/)build.*',
-            r'^\.(\\|/)bin.*', r'.*readthedocs.*']
+            r'^\.(\\|/)bin.*', r'.*readthedocs.*',
+            r'.*venv.*', r'.*cmake-build-.*']
```
Comment (Myoldmopar) [inline scripts/dev/license-check.py:105]: The license check script was throwing warnings about two directories:
- `repo_root/venv`: an exceedingly common place for a Python virtual environment
- `repo_root/cmake-build-debug`: where CLion creates a build folder -- also can be `cmake-build-release` and others, so just matching on the base pattern.

File: src/EnergyPlus/ConvectionCoefficients.cc [hunk 5]
```diff
@@ -3175,8 +3175,7 @@ Real64 CalcASHRAETARPNatural(Real64 const Tsurf, Real64 const Tamb, Real64 const
 
         return CalcWaltonUnstableHorizontalOrTilt(DeltaTemp, cosTilt);
 
-    } else /*if (((DeltaTemp > 0.0) && (cosTilt < 0.0)) ||
-               ((DeltaTemp < 0.0) && (cosTilt > 0.0)))*/
+    } else // if (((DeltaTemp > 0.0) && (cosTilt < 0.0)) || ((DeltaTemp < 0.0) && (cosTilt > 0.0)))
```
Comment (Myoldmopar) [inline src/EnergyPlus/ConvectionCoefficients.cc:3178]: Just converted to a single line comment after the `else`

**Review by Myoldmopar — COMMENTED** [review]
Code walkthrough.  Not too much to see really, converted the few C style comments to C++ style and added a script to have GHA warn on if any are introduced.

**Comment by Myoldmopar** [comment]
If anyone has concerns with this, speak up.  Based on the positive response on Slack, I don't envision anyone will.  Once CI is all done this should merge and get out of the way for other PRs.

**Comment by amirroth** [comment]
Maybe in exchange for removing C-style comments from the source code, we can introduce them into epJSON.

**Comment by Myoldmopar** [comment]
Maybe in exchange for this we can go back to using tabs instead of spaces
giphy

Just kidding!  OK, merging this and locking the conversation before it even gets started :smile:

**Pull request merged**